---
title: "Vizgen Mouse Brain Receptor Map"
author: "Rafael Peixoto"
date: "9/29/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
agg_mat_preview = data.frame(V1 = c('110883424764611924400221639916314253469',
                                    '135188247894899244046039873973964001182',
                                    '164766962839370328502017156371562646881',
                                    '165747897693809971960756442245389760838'),
                             Oxgr1 = c(0,0,0,0),
                             Htr1a = c(0,0,0,0),
                             Htr1b = c(0,0,0,1))
cell_met_preview = data.frame(V1 = c('110883424764611924400221639916314253469',
                                     '135188247894899244046039873973964001182',
                                     '164766962839370328502017156371562646881',
                                     '165747897693809971960756442245389760838'),
                              fov = c(0,0,0,0),
                              volume = c(432.1414, 1351.8026, 1080.6533, 1652.0007),
                              center_x = c(156.5633, 156.5093, 159.9653, 167.5793),
                              center_y = c(4271.326, 4256.962, 4228.180, 4323.868),
                              min_x = c(151.5305, 148.2905, 152.1785, 158.2265),
                              max_x = c(161.5961, 164.7281, 167.7521, 176.9321),
                              min_y = c(4264.620, 4247.664, 4220.556, 4314.192),
                              max_y = c(4278.033, 4266.261, 4235.805, 4333.545))
tx_det_preview = data.frame(V1 = c(0,1,2,3),
                            barcode_id = c(2,2,11,13),
                            global_x = c(159.9778, 165.9403, 158.4767, 171.2179),
                            global_y = c(4208.468, 4321.805, 4320.901, 4283.950),
                            global_z = c(4,4,5,0),
                            x = c(1762.116, 1817.324, 1748.217, 1866.191),
                            y = c(159.9814, 1209.3922, 1201.0283, 858.8855),
                            fov = c(0,0,0,0),
                            gene = c('Htr1b', 'Htr1b', 'Htr6', 'Adora1'))
transform_DT = data.table::data.table(V1 = c(9.205861,0,0),
                                      V2 = c(0,9.20585,0),
                                      V3 = c(279.2204,349.8105,1.0000))
km_spatialgenes = list(feats = c("Slc47a1","Chat","Th","Insrr","Slc17a7","Pln","Lmod1","Blank-119",
                       "Hcar1","Glp1r","Ptgdr","Avpr2","Gpr20","Myh11","Glp2r","Npy2r","Gpr182",
                       "Chrm1","Adgrd1","Mrgprf","Trhr","Gfap","Slc17a8","Nmbr","Pth2r","Rxfp1",
                       "Musk","F2rl1","Dgkk","Chrm5"))
```

Updated `9/12/2022`. *Giotto* version `2.0.0.998`.

Please check the version you are using to get the same results.

## 1. Set up Giotto

```{r, eval=FALSE}
library(Giotto)

# 1. set working directory where project outputs will be saved to
results_folder = '/path/to/save/directory/'

# 2. set giotto python path
# set python path to your preferred python version path
# set python path to NULL if you want to automatically install (only the 1st time) and use the giotto miniconda environment
python_path = NULL 
if(is.null(python_path)) {
  installGiottoEnvironment()
}
```

## 2. Dataset explanation

This vignette covers Giotto object creation and exploratory analysis with the subcellular [MERFISH Mouse Brain Receptor Map dataset](https://vizgen.com/applications/neuroscience-showcase/) provided by Vizgen with their MERSCOPE platform.

Transcripts are captured at the single-molecule level with subcellular spatial resolution (â‰¤100nm). This dataset includes information from 9 full coronal mouse brain slices (3 positions with 3 biological replicates) that were profiled for 483 genes. This vignette works with slice 1, replicate 1.

**Provided Outputs:**

-   List of all **detected transcripts** and their spatial locations in three dimensions (CSV)

    <details>
    <summary>Show first 4 rows</summary>
    ```{r, echo=FALSE}
    print(tx_det_preview)
    ```
    </details>

-   Output from the cell segmentation analysis:

    -   **transcripts** (cols) per **cell** (rows) aggregated count matrix (CSV)

        <details>
        <summary>Show first 4 rows and columns</summary>
        ```{r, echo=FALSE}
        print(agg_mat_preview)
        ```
        </details>

    -   **cell metadata** (CSV)
    
        <details>
        <summary>Show first 4 rows</summary>
        ```{r, echo=FALSE}
        print(cell_met_preview)
        ```
        </details>

    -   **cell boundaries** (HDF5)

-   The **DAPI** and **Poly T mosaic images** (TIFF)

`Vizgen Data Release V1.0. May 2021`

## 3. Giotto global instructions and preparations

Define plot saving behavior and project data paths

```{r, eval=FALSE}
# Directly saving plots to the working directory without rendering them in the editor saves time.
instrs = createGiottoInstructions(save_dir = results_folder,
                                  save_plot = TRUE,
                                  show_plot = FALSE,
                                  return_plot = FALSE)

# Add Needed paths below:
# provide path to pre-aggregated information
expr_path = '/path/to/datasets_mouse_brain_map_BrainReceptorShowcase_Slice1_Replicate1_cell_by_gene_S1R1.csv'
# provide path to metadata (includes spatial locations of aggregated expression)
meta_path = '/path/to/datasets_mouse_brain_map_BrainReceptorShowcase_Slice1_Replicate1_cell_metadata_S1R1.csv'
# provide path to the detected transcripts (single molecule level transcript spatial information)
tx_path = '/path/to/datasets_mouse_brain_map_BrainReceptorShowcase_Slice1_Replicate1_detected_transcripts_S1R1.csv'
# provide path to the dapi image of slice 1 replicate 1
img_path = 'path/to/mosaic_DAPI_z0.tif'
```

## 4. Create Giotto object from aggregated data

Vizgen provides a cell by transcript output matrix (`cell_by_gene.csv`) with the subcellular spatial transcript information already aggregated by the provided polygon cell annotations into a count matrix.  
Along with the count matrix, metadata information about the field of view (FOV), spatial location, and volume of the cell (annotation polygons) is also provided through the `cell_metadata.csv`.  

**Pre-aggregated** information can be loaded into Giotto with the usual generic `createGiottoObject()` function. For starting from the raw subcellular information, skip to [step 10](). To create the Giotto object, the `cell_by_gene` expression matrix and the `cell_metadata` information are first read into R. Since Giotto accepts the expression information with features (in this case genes/transcript counts) as rows and cells as columns, the expression matrix must first be transposed to create the object.  
  
*Addtionally for this dataset, y values should be inverted when loaded to match the included images. [For more information](https://giottosuite.com/articles/getting_started_images.html#standard-workflow)

```{r, eval=FALSE}
# read expression matrix and metadata
expr_matrix = readExprMatrix(expr_path)
spat_dt = data.table::fread(meta_path)

# create giotto object
vizgen <- createGiottoObject(expression = Giotto:::t_flex(expr_matrix),
                             spatial_locs = spat_dt[,.(center_x, -center_y, V1)],
                             instructions = instrs)
# add metadata of fov and volume
vizgen <- addCellMetadata(vizgen,
                          new_metadata = spat_dt[,.(fov, volume)])
```

### Visualize cells in space.

```{r, eval=FALSE}
spatPlot2D(vizgen,
           point_size = 0.5)
```

![](../inst/images/vizgen_mouse_brain//0-overview_spatplot.png){width="60%"}

### Visualize cells by FOV.

```{r fovplot, eval=FALSE}
spatPlot2D(vizgen, point_size = 0.5,
           cell_color = 'fov',
           show_legend = F)
```

![](../inst/images/vizgen_mouse_brain//1-FOVs.png){width="60%"}

## 5. Attaching images

Images for confocal planes z0 to z6 are provided for both **DAPI** (cell nucleus staining) and **polyT** for all datasets.  
  A `micron_to_mosaic_pixel_transform.csv` is included within the `images` folder that provides scaling factors to map the image to the spatial coordinates. For this dataset:  
<details>
<summary>micron_to_mosaic_pixel_transform.csv</summary>
```{r, echo=FALSE}
print(transform_DT)
```
</details>
<br>
Here we will attach the z0 dapi image to the Giotto object.
Note: *It is recommended for the image files to be local. Placing the images on the cloud or network may be very slow.*
```{r, eval=FALSE}
# Load in image as a giottoLargeImage object that maps spatial coordinates 1:1 with pixel coordinates
dapi0 = createGiottoLargeImage(raster_object = img_path,
                               name = 'image')

# Preview image
plot(dapi0)
```

![](../inst/images/vizgen_mouse_brain//2-dapi0_preview.png){width="65%"}

Attaching the `giottoLargeImage` to our Giotto object (provided as a list of 1) and then updating it to map the image to the spatial coordinates which are in microns.   
  
```{r, eval=FALSE}
# Adds the giottoLargeImage object to giotto object while also shifting values into the negatives
vizgen = addGiottoImage(gobject = vizgen,
                        largeImages = list(dapi0),
                        negative_y = TRUE)

img_scale_path = 'path/to/micron_to_mosaic_pixel_transform.csv'

img_scale_DT = data.table::fread(img_scale_path)
x_scale = img_scale_DT$V1[[1]]
y_scale = img_scale_DT$V2[[2]]
x_shift = img_scale_DT$V3[[1]]
y_shift = -img_scale_DT$V3[[2]]

# Update image to reverse the above transformations to convert mosaic pixel to micron
# 'first_adj' means that the xy shifts are applied before the subsequent scaling
vizgen = updateGiottoLargeImage(gobject = vizgen,
                                largeImage_name = 'image',
                                x_shift = -x_shift,
                                y_shift = -y_shift,
                                scale_x = 1/x_scale,
                                scale_y = 1/y_scale,
                                order = 'first_adj')
```

### Check image alignment
```{r, eval=FALSE}
spatPlot2D(gobject = vizgen,
           largeImage_name = 'image',
           point_size = 0.5,
           show_image = TRUE)
```
![](../inst/images/vizgen_mouse_brain//3-img_overlay.png){width="60%"}

### Zooming in by subsetting the dataset
```{r, eval=FALSE}
zoom = subsetGiottoLocs(gobject = vizgen,
                        x_min = 2000,
                        x_max = 2500,
                        y_min = -2500,
                        y_max = -2000)

spatPlot2D(gobject = zoom,
           largeImage_name = 'image',
           point_size = 1,
           show_image = TRUE)
```
![](../inst/images/vizgen_mouse_brain//4-img_overlay_zoom.png){width="60%"}

## 6. Data processing

```{r, eval=FALSE}
vizgen <- filterGiotto(gobject = vizgen,
                       expression_threshold = 1,
                       feat_det_in_min_cells = 100,
                       min_det_feats_per_cell = 20)

vizgen <- normalizeGiotto(gobject = vizgen,
                          scalefactor = 1000,
                          verbose = TRUE)

# add gene and cell statistics
vizgen <- addStatistics(gobject = vizgen)
```

### Visualize the number of features per cell.

```{r, eval=FALSE}
spatPlot2D(gobject = vizgen,
           show_image = F,
           point_alpha = 0.7,
           cell_color = 'nr_feats',
           color_as_factor = F,
           point_border_col = 'grey',
           point_border_stroke = 0.01,
           point_size = 0.5)
```

![](../inst/images/vizgen_mouse_brain//5-nr_feats.png){width="60%"}

## 7. Dimension reduction

Skipping highly variable feature (HVF) detection. PCA will be calculated based on all available genes.
```{r, eval=FALSE}
vizgen <- runPCA(gobject = vizgen,
                 center = TRUE,
                 scale_unit = TRUE)
# visualize variance explained per component 
screePlot(vizgen,
          ncp = 30)
```

![](../inst/images/vizgen_mouse_brain//6-screePlot.png){width="50%"}

```{r, eval=FALSE}
plotPCA(gobject = vizgen,
        point_size = 0.5)
```

![](../inst/images/vizgen_mouse_brain//7-PCA.png){width="50%"}

```{r, eval=FALSE}
vizgen = runUMAP(vizgen,
                 dimensions_to_use = 1:10)

plotUMAP(gobject = vizgen,
         point_size = 0.5)
```

![](../inst/images/vizgen_mouse_brain//8-UMAP.png){width="50%"}

## 8. Leiden Clustering

Calculate nearest neighbor network and perform Leiden clustering.

```{r, eval=FALSE}
vizgen <- createNearestNetwork(vizgen,
                               dimensions_to_use = 1:10,
                               k = 15)

vizgen <- doLeidenCluster(vizgen,
                          resolution = 0.2,
                          n_iterations = 100)
```

Visualize clusters in reduced dimension. The default cell color is 'leiden_clus'.

```{r, eval=FALSE}
plotUMAP(vizgen,
         cell_color = 'leiden_clus',
         point_size = 0.5)
```

![](../inst/images/vizgen_mouse_brain//9-UMAP.png){width="50%"}

Visualize in spatial dimensions.

```{r, eval=FALSE}
spatPlot2D(gobject = vizgen,
           cell_color = 'leiden_clus',
           point_size = 0.5,
           background_color = 'black')
```
![](../inst/images/vizgen_mouse_brain//10-spatPlot2D.png){width="60%"}


## 9. Spatial expression patterns

Spatially interesting gene expression can be detected by first generating a spatial network then performing Binary Spatial Extraction of genes.

```{r, eval=FALSE}
# create spatial network based on physical distance of cell centroids
vizgen = createSpatialNetwork(gobject = vizgen,
                              minimum_k = 2,
                              maximum_distance_delaunay = 50)

# perform Binary Spatial Extraction of genes
km_spatialgenes = binSpect(vizgen)
```

```{r, echo=TRUE}
print(km_spatialgenes$feats[1:30])
```

```{r, eval=FALSE}
# visualize spatial expression of select genes obtained from binSpect
spatFeatPlot2D(vizgen,
               expression_values = 'scaled',
               feats = km_spatialgenes$feats[c(1,2,3,5,18,22)],
               cell_color_gradient = c('blue', 'white', 'red'),
               point_shape = 'border',
               point_border_col = 'grey',
               point_border_stroke = 0.01,
               point_size = 0.2,
               cow_n_col = 2)
```

![](../inst/images/vizgen_mouse_brain//11-spatFeatPlot2D.png){width="60%"}

## 10. Working with subcellular information

Subsetting to work with a smaller area since subcellular data is dense.

```{r, eval=FALSE}
vizgen_subset <- subsetGiottoLocs(gobject = vizgen,
                                  x_min = 2000, x_max = 3000,
                                  y_max = -2500, y_min = -3500)
```

The visualization functions can also be applied to the subset version.

```{r, eval=FALSE}
spatPlot2D(gobject = vizgen_subset,
           largeImage_name = 'image',
           cell_color = 'leiden_clus',
           point_size = 2.5)
```

![](../inst/images/vizgen_mouse_brain//12-spatPlot2D.png){width="50%"}

These steps may require a strong computer.  
Giotto can also work with the polygon annotations provided by Vizgen. Since we are working with a subset of the data, it is only necessary to read the polygons that are present in the current FOV(s).

```{r, eval=FALSE}
# define path to cell boundaries folder
bound_path = '/path/to/cell_boundaries'
# read polygons and add them to Giotto
vizgen_subset = readPolygonFilesVizgen(gobject = vizgen_subset,
                                       boundaries_path = bound_path,
                                       polygon_feat_types = c(0,4,6))
```

Giotto can also work with the single molecule transcript level spatial data.

```{r, eval=FALSE}
tx_dt = data.table::fread(tx_path)
# select transcripts in FOVs
selected_fovs = unique(pDataDT(vizgen_subset)$fov)
tx_dt_selected = tx_dt[fov %in% selected_fovs]

# create Giotto points from transcripts
gpoints = createGiottoPoints(x = tx_dt_selected[,.(global_x,-global_y, gene)])

# add points to Giotto
vizgen_subset = addGiottoPoints(gobject = vizgen_subset,
                                gpoints = list(gpoints))

# identify genes for visualization
gene_meta = fDataDT(vizgen_subset)
data.table::setorder(gene_meta, perc_cells)
gene_meta[perc_cells > 25 & perc_cells < 50]

# visualize points with z0 polygons (confocal plane)
spatInSituPlotPoints(vizgen_subset,
                     feats = list('rna' = c("Oxgr1", "Htr1a", "Gjc3", "Axl",
                                            'Gfap', "Olig1", "Epha7")),
                     polygon_feat_type = 'z0',
                     use_overlap = F,
                     point_size = 0.2,
                     show_polygon = TRUE,
                     polygon_color = 'white')
```

![](../inst/images/vizgen_mouse_brain//11-spatInSituPlotPoints.png){width="50%"}

```{r, eval=FALSE}
# visualize points with polygons from z6 (confocal plane)
spatInSituPlotPoints(vizgen_subset,
                     feats = list('rna' = c("Oxgr1", "Htr1a", "Gjc3", "Axl",
                                            'Gfap', "Olig1", "Epha7")),
                     polygon_feat_type = 'z6',
                     use_overlap = F,
                     point_size = 0.2,
                     show_polygon = TRUE,
                     polygon_color = 'white')
```

![](../inst/images/vizgen_mouse_brain//12-spatInSituPlotPoints.png){width="50%"}



