<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> â€¢ Giotto Suite</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="Giotto">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-BCXT5392QC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-BCXT5392QC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto Suite</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0.997</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Start
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/getting_started.html#installation">Installation</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#getting-started">Getting Started</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#analyses">Analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/datasets_overview.html#array-based-datasets-1">Array based</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#single-cell-1">Single-cell</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#subcellular-datasets-1">Subcellular</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#protein-multiplexing-1">Protein multiplexing</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/contribute.html">Contribute</a>
</li>
<li>
  <a href="../articles/giotto_news.html">News</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/faqs.html">Frequently Asked Questions</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">GitHub Issues Guidelines</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Try Giotto
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Docker</li>
    <li class="dropdown-header">Binder</li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RubD/Giotto/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="getting_started_gobject_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RubD/Giotto/blob/master/vignettes/getting_started_gobject.Rmd"><code>vignettes/getting_started_gobject.Rmd</code></a></small>
      <div class="hidden name"><code>getting_started_gobject.Rmd</code></div>

    </div>

    
    
<div id="how-to-create-a-giotto-object" class="section level1">
<h1 class="hasAnchor">
<a href="#how-to-create-a-giotto-object" class="anchor"></a>How to create a Giotto Object?</h1>
<div id="minimum-requirements" class="section level4">
<h4 class="hasAnchor">
<a href="#minimum-requirements" class="anchor"></a>1. Minimum requirements</h4>
<ul>
<li>expression matrix</li>
<li>spatial locations (<em>unnecessary for scRNAseq analysis</em>)</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rubd.github.io/Giotto/">Giotto</a></span><span class="op">)</span>

<span class="co"># 1. Directly from file path (dummy locations are used as examples here)</span>
<span class="va">path_to_expression_file</span> <span class="op">=</span> <span class="st">"/local/path/to/expression/file.txt"</span>
<span class="va">path_to_locations_file</span> <span class="op">=</span> <span class="st">"/local/path/to/locations/file.txt"</span>

<span class="va">my_giotto_object</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoObject.html">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">path_to_expression_file</span>,
                                      spatial_locs <span class="op">=</span> <span class="va">path_to_locations_file</span><span class="op">)</span>

<span class="co"># 2. Use objects already loaded into environment</span>
<span class="va">expression_matrix</span> <span class="op">=</span> <span class="fu"><a href="../reference/readExprMatrix.html">readExprMatrix</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">path_to_expression_file</span><span class="op">)</span> <span class="co"># fast method to read expression matrix</span>
<span class="va">cell_locations</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">path_to_locations_file</span><span class="op">)</span>

<span class="va">my_giotto_object</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoObject.html">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expression_matrix</span>,
                                      spatial_locs <span class="op">=</span> <span class="va">cell_locations</span><span class="op">)</span></code></pre></div>
<details><summary>
Expression file formatting
</summary><p><br> Expression count matrices accepted by Giotto should have columns correlating to individual cells (<strong>cell IDs</strong>) and rows to individual features (<strong>feature IDs</strong>).<br><strong>data.table</strong> objects and text-based files ie: (<strong>.csv, .tsv, .txt</strong>) to be read in should have column 1 be <strong>feature IDs</strong>.</p>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="left">feat_ID</th>
<th>Cell_1</th>
<th>Cell_2</th>
<th>Cell_3</th>
<th>â€¦</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td align="left">Gene1</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td align="left">Gene2</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td align="left">Gene3</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td align="left">â€¦</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p><br><strong>matrix</strong> and <strong>DelayedMatrix</strong> objects should have <strong>feature IDs</strong> already incorporated as the rownames.</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th>Cell_1</th>
<th>Cell_2</th>
<th>Cell_3</th>
<th>â€¦</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Gene1</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td align="left">Gene2</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td align="left">Gene3</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td align="left">â€¦</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table></details><details><summary>
Locations file formatting
</summary><p><br> Numerical columns will be interpreted in order of x, y, and (optionally) z coordinate. The first non-numerical column will be taken as <strong>cell IDs</strong></p>
<table class="table">
<thead><tr class="header">
<th></th>
<th>sdimx</th>
<th>sdimy</th>
<th>sdimz</th>
<th>cell_ID</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table></details><p><strong>*Note:</strong> multiple expression files during can be given at once during Giotto object creation by using a named list.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>my_giotto_object <span class="ot">=</span> <span class="fu">createGiottoObject</span>(<span class="at">expression =</span> <span class="fu">list</span>(<span class="at">raw =</span> matrix1</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">scaled =</span> matrix2,</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">custom =</span> matrix3),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">spatial_locs =</span> cell_locations)</span></code></pre></div>
<p><br></p>
</div>
<div id="additional-features" class="section level4">
<h4 class="hasAnchor">
<a href="#additional-features" class="anchor"></a>2. Additional features</h4>
<p>Other information can also be provided using the other function parameters<br>
- add <strong>cell</strong> or <strong>gene metadata</strong><br>
- add <strong>spatial networks</strong> or <strong>grids</strong><br>
- add <strong>dimension reduction</strong><br>
- add <strong>images</strong><br>
- add <strong>instructions</strong></p>
<p>Usually specifying your own <strong>instructions</strong> can be most useful to:<br>
- specify a python path<br>
- determine output of plots (show in viewer, return as object, save as image)<br>
- automatically save plots to selected directory</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rubd.github.io/Giotto/">Giotto</a></span><span class="op">)</span>

<span class="co"># 1. Directly from file path (dummy locations are used as examples here)</span>
<span class="va">path_to_expression_file</span> <span class="op">=</span> <span class="st">"/local/path/to/expression/file.txt"</span>
<span class="va">path_to_locations_file</span> <span class="op">=</span> <span class="st">"/local/path/to/locations/file.txt"</span>

<span class="co"># 2. Create your own instructions</span>
<span class="va">path_to_python</span> <span class="op">=</span> <span class="st">'/usr/bin/python3'</span> <span class="co"># can be something else</span>
<span class="va">working_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># this will use your current working directory</span>
<span class="va">my_instructions</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoInstructions.html">createGiottoInstructions</a></span><span class="op">(</span>python_path <span class="op">=</span> <span class="va">path_to_python</span>,
                                           save_dir <span class="op">=</span> <span class="va">working_directory</span><span class="op">)</span>

<span class="co"># 3. Create your giotto object</span>
<span class="va">my_giotto_object</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoObject.html">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">path_to_expression_file</span>,
                                      spatial_locs <span class="op">=</span> <span class="va">path_to_locations_file</span>,                                      
                                      cell_metadata <span class="op">=</span> <span class="va">my_cell_metadata</span>,
                                      gene_metadata <span class="op">=</span> <span class="va">my_gene_metadata</span>,
                                      instructions <span class="op">=</span> <span class="va">my_instructions</span><span class="op">)</span>

<span class="co"># 4. check which giotto instructions are associated with your giotto object</span>
<span class="fu"><a href="../reference/showGiottoInstructions.html">showGiottoInstructions</a></span><span class="op">(</span><span class="va">my_giotto_object</span><span class="op">)</span></code></pre></div>
<p><br></p>
</div>
</div>
<div id="giotto-object-structure" class="section level1">
<h1 class="hasAnchor">
<a href="#giotto-object-structure" class="anchor"></a>Giotto Object Structure</h1>
<p><br> Usage of the Giotto package revolves around the <strong>giottoObject</strong>. This is an S4 object class that holds spatial expression data and facilitates its use with the Giotto packageâ€™s functions. Additional metadata and other certain outputs generated from Giotto functions that are used in downstream analyses will also be stored within this object. The self-contained nature of the Giotto object makes it a convenient representation of the entire spatial experiment and is why many of Giotto functions both take the <strong>giottoObject</strong> as input and feed into it as output.</p>
<p>Data is organized within the <strong>giottoObject</strong> in defined <strong>slots</strong> as described in the diagram below.<br><br><img src="../inst/images/getting_started_figs/getting_started_gobject/Giotto_suite_object-01.svg" style="width:100.0%"><br><img src="../inst/images/getting_started_figs/getting_started_gobject/Giotto_suite_object-02.svg" style="width:100.0%"><br><img src="../inst/images/getting_started_figs/getting_started_gobject/Giotto_suite_object-03.svg" style="width:100.0%"></p>
<div id="nested-organization-within-slots" class="section level2">
<h2 class="hasAnchor">
<a href="#nested-organization-within-slots" class="anchor"></a>Nested organization within slots</h2>
<p>Spatial data has a tiered organization and so does Giotto. Tissue regions can be divided down into individual cells and then further into organelles. The data structure of Giotto mirrors biological structure by nesting data within slots related to spatial data by <strong>spat_unit</strong> (spatial unit). Additonally, to facilitate the integration of multiple omic data, slots related to feature data will be nested with <strong>feat_type</strong> (feature type). <br><br>
Included below is a description of this nesting and also, for advanced users and contributors/developers, accessor functions for the slots are also provided. <br></p>
<div id="slots-and-subnesting" class="section level3">
<h3 class="hasAnchor">
<a href="#slots-and-subnesting" class="anchor"></a>Slots and Subnesting</h3>
<table class="table">
<thead><tr class="header">
<th align="left">Slot</th>
<th align="left">Nested</th>
<th align="left">Example</th>
<th align="left">Accessors</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong><span class="citation">@expression</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #E88EBB;">feat_type</span>-<span style="color: #801C34;">name</span>
</td>
<td align="left">
<span style="color: #4C94A0;">cell</span>-<span style="color: #E88EBB;">rna</span>-<span style="color: #801C34;">raw</span>
</td>
<td align="left">get_expression_values()<br>set_expression_values()<br>showGiottoExpression()</td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@cell_metadata</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #E88EBB;">feat_type</span>
</td>
<td align="left">
<span style="color: #4C94A0;">cell</span>-<span style="color: #E88EBB;">rna</span>
</td>
<td align="left">pDataDT()<br>addCellMetadata()</td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@feat_metadata</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #E88EBB;">feat_type</span>
</td>
<td align="left">
<span style="color: #4C94A0;">cell</span>-<span style="color: #E88EBB;">rna</span>
</td>
<td align="left">fDataDT()<br>addFeatMetadata()</td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@spatial_grid</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #801C34;">name</span>
</td>
<td align="left">
<span style="color: #4C94A0;">grid</span>-<span style="color: #801C34;">grid</span>
</td>
<td align="left">get_spatialGrid()<br>set_spatialGrid()<br>showGiottoSpatGrids()</td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@nn_network</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #F6851F;">method</span>-<span style="color: #801C34;">name</span>
</td>
<td align="left">
<span style="color: #4C94A0;">cell</span>-<span style="color: #F6851F;">sNN</span>-<span style="color: #801C34;">sNN_results1</span>
</td>
<td align="left">get_NearestNetwork()<br>set_NearestNetwork()</td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@dimension_reduction</span></strong></td>
<td align="left">
<span style="color: #6C04D0;">approach</span>-<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #E88EBB;">feat_type</span>-<span style="color: #F6851F;">method</span>-<span style="color: #801C34;">name</span>
</td>
<td align="left">
<span style="color: #6C04D0;">cells</span>-<span style="color: #4C94A0;">cell</span>-<span style="color: #E88EBB;">rna</span>-<span style="color: #F6851F;">pca</span>-<span style="color: #801C34;">pca</span>
</td>
<td align="left">get_dimReduction()<br>set_dimReduction()<br>showGiottoDimRed()</td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@spatial_enrichment</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #E88EBB;">feat_type</span>-<span style="color: #801C34;">name</span>
</td>
<td align="left">
<span style="color: #4C94A0;">cell</span>-<span style="color: #E88EBB;">rna</span>-<span style="color: #801C34;">results1</span>
</td>
<td align="left">get_spatial_enrichment()<br>set_spatial_enrichment()<br>showGiottoSpatEnrichments()</td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@spatial_info</span></strong></td>
<td align="left"><span style="color: #4C94A0;">spat_unit</span></td>
<td align="left"><span style="color: #4C94A0;">cell</span></td>
<td align="left">get_polygon_info()<br>set_polygon_info()<br>showGiottoSpatialInfo()</td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@spatial_locs</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #801C34;">name</span>
</td>
<td align="left">
<span style="color: #4C94A0;">cell</span>-<span style="color: #801C34;">raw</span>
</td>
<td align="left">get_spatial_locations()<br>set_spatial_locations()<br>showGiottoSpatLocs()</td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@spatial_network</span></strong></td>
<td align="left">
<span style="color: #4C94A0;">spat_unit</span>-<span style="color: #801C34;">name</span>
</td>
<td align="left">
<span style="color: #4C94A0;">cell</span>-<span style="color: #801C34;">Delaunay_network1</span>
</td>
<td align="left">get_spatialNetwork()<br>set_spatialNetwork()<br>showGiottoSpatNetworks()</td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@feat_info</span></strong></td>
<td align="left"><span style="color: #E88EBB;">feat_type</span></td>
<td align="left"><span style="color: #E88EBB;">rna</span></td>
<td align="left">get_feature_info()<br>set_feature_info()<br>showGiottoFeatInfo()</td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@images</span></strong></td>
<td align="left"><span style="color: #801C34;">name</span></td>
<td align="left"><span style="color: #801C34;">image</span></td>
<td align="left">getGiottoImage()<br>addGiottoImage()<br>showGiottoImageNames()</td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@largeImages</span></strong></td>
<td align="left"><span style="color: #801C34;">name</span></td>
<td align="left"><span style="color: #801C34;">image</span></td>
<td align="left">getGiottoImage()<br>addGiottoImage()<br>showGiottoImageNames()</td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@instructions</span></strong></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">replaceGiottoInstructions()<br>showGiottoInstructions()</td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@OS_platform</span></strong></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"><strong><span class="citation">@join_info</span></strong></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"><strong><span class="citation">@parameters</span></strong></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.drieslab.com/">Ruben Dries</a>, Jiaji George Chen, Natalie Del Rossi, Emma Kelley, Junxiang Xu, Guo-Cheng Yuan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
