<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>mini_seqfish â€¢ Giotto Suite</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="mini_seqfish">
<meta property="og:description" content="Giotto">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-BCXT5392QC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-BCXT5392QC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto Suite</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0.991</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Start
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/getting_started.html#installation">Installation</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#getting-started">Getting Started</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#analyses">Analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/datasets_overview.html#array-based-datasets-1">Array based</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#single-cell-1">Single-cell</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#subcellular-datasets-1">Subcellular</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#protein-multiplexing-1">Protein multiplexing</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/contribute.html">Contribute</a>
</li>
<li>
  <a href="../articles/giotto_news.html">News</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/faqs.html">Frequently Asked Questions</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">GitHub Issues Guidelines</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Try Giotto
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Docker</li>
    <li class="dropdown-header">Binder</li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RubD/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>mini_seqfish</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RubD/Giotto/blob/HEAD/vignettes/mini_seqfish.Rmd" class="external-link"><code>vignettes/mini_seqfish.Rmd</code></a></small>
      <div class="hidden name"><code>mini_seqfish.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rubd.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="install-python-modules">Install Python modules<a class="anchor" aria-label="anchor" href="#install-python-modules"></a>
</h3>
<p>To run this vignette you need to install all the necessary Python modules.</p>
<ol style="list-style-type: decimal">
<li><p>This can be done manually, see <a href="https://rubd.github.io/Giotto_site/articles/installation_issues.html#python-manual-installation" class="external-link uri">https://rubd.github.io/Giotto_site/articles/installation_issues.html#python-manual-installation</a></p></li>
<li><p>This can be done within R using our installation tools (installGiottoEnvironment), see <a href="https://rubd.github.io/Giotto_site/articles/tut0_giotto_environment.html" class="external-link uri">https://rubd.github.io/Giotto_site/articles/tut0_giotto_environment.html</a> for more information.</p></li>
</ol>
</div>
<div class="section level3">
<h3 id="optional-set-giotto-instructions">(optional) set giotto instructions<a class="anchor" aria-label="anchor" href="#optional-set-giotto-instructions"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># to automatically save figures in save_dir set save_plot to TRUE</span>
<span class="va">temp_dir</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#temp_dir = '~/Temp/'</span>
<span class="va">myinstructions</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoInstructions.html">createGiottoInstructions</a></span><span class="op">(</span>save_dir <span class="op">=</span> <span class="va">temp_dir</span>,
                                          save_plot <span class="op">=</span> <span class="cn">FALSE</span>, 
                                          show_plot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-giotto-object">1. Create a Giotto object<a class="anchor" aria-label="anchor" href="#create-a-giotto-object"></a>
</h2>
<p>minimum requirements:<br>
- matrix with expression information (or path to)<br>
- x,y(,z) coordinates for cells or spots (or path to)</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># giotto object </span>
<span class="va">expr_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"seqfish_field_expr.txt.gz"</span>, package <span class="op">=</span> <span class="st">'Giotto'</span><span class="op">)</span>
<span class="va">loc_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"seqfish_field_locs.txt"</span>, package <span class="op">=</span> <span class="st">'Giotto'</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createGiottoObject.html">createGiottoObject</a></span><span class="op">(</span>raw_exprs <span class="op">=</span> <span class="va">expr_path</span>,
                                   spatial_locs <span class="op">=</span> <span class="va">loc_path</span>,
                                   instructions <span class="op">=</span> <span class="va">myinstructions</span><span class="op">)</span></code></pre></div>
<p>How to work with Giotto instructions that are part of your Giotto object:<br>
- show the instructions associated with your Giotto object with <strong>showGiottoInstructions</strong><br>
- change one or more instructions with <strong>changeGiottoInstructions</strong><br>
- replace all instructions at once with <strong>replaceGiottoInstructions</strong><br>
- read or get a specific giotto instruction with <strong>readGiottoInstructions</strong><br>
Of note, the python path can only be set once in an R session. See the reticulate package for more information.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show instructions associated with giotto object (seqfish_mini)</span>
<span class="fu"><a href="../reference/showGiottoInstructions.html">showGiottoInstructions</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="processing-steps">2. processing steps<a class="anchor" aria-label="anchor" href="#processing-steps"></a>
</h2>
<ul>
<li>filter genes and cells based on detection frequencies<br>
</li>
<li>normalize expression matrix (log transformation, scaling factor and/or z-scores)</li>
<li>add cell and gene statistics (optional)<br>
</li>
<li>adjust expression matrix for technical covariates or batches (optional). These results will be stored in the <em>custom</em> slot.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, 
                             expression_threshold <span class="op">=</span> <span class="fl">0.5</span>, 
                             gene_det_in_min_cells <span class="op">=</span> <span class="fl">20</span>, 
                             min_det_genes_per_cell <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, scalefactor <span class="op">=</span> <span class="fl">6000</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjustGiottoMatrix.html">adjustGiottoMatrix</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, 
                                   expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'normalized'</span><span class="op">)</span>,
                                   covariate_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'nr_feats'</span>, <span class="st">'total_expr'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dimension-reduction">3. dimension reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<ul>
<li>identify highly variable genes (HVG)<br>
</li>
<li>perform PCA<br>
</li>
<li>identify number of significant prinicipal components (PCs)<br>
</li>
<li>run UMAP and/or TSNE on PCs (or directly on matrix)</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu">calculateHVG</span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span>
<span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, ncp <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="fu"><a href="../reference/jackstrawPlot.html">jackstrawPlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, ncp <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotPCA.html">plotPCA</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span>

<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, n_threads <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runtSNE.html">runtSNE</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTSNE.html">plotTSNE</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="clustering">4. clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<ul>
<li>create a shared (default) nearest network in PCA space (or directly on matrix)<br>
</li>
<li>cluster on nearest network with Leiden or Louvan (kmeans and hclust are alternatives)</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, k <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, resolution <span class="op">=</span> <span class="fl">0.4</span>, n_iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>

<span class="co"># visualize UMAP cluster results</span>
<span class="fu"><a href="../reference/plotUMAP.html">plotUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, show_NN_network <span class="op">=</span> <span class="cn">T</span>, point_size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span>

<span class="co"># visualize UMAP and spatial results</span>
<span class="fu"><a href="../reference/spatDimPlot.html">spatDimPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span>, spat_point_shape <span class="op">=</span> <span class="st">'voronoi'</span><span class="op">)</span>

<span class="co"># heatmap and dendrogram</span>
<span class="fu"><a href="../reference/showClusterHeatmap.html">showClusterHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span>
<span class="fu"><a href="../reference/showClusterDendrogram.html">showClusterDendrogram</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, h <span class="op">=</span> <span class="fl">0.5</span>, rotate <span class="op">=</span> <span class="cn">T</span>, cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="differential-expression">5. differential expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gini_markers</span> <span class="op">=</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                                                  method <span class="op">=</span> <span class="st">'gini'</span>,
                                                  expression_values <span class="op">=</span> <span class="st">'normalized'</span>,
                                                  cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span>,
                                                  min_genes <span class="op">=</span> <span class="fl">20</span>,
                                                  min_expr_gini_score <span class="op">=</span> <span class="fl">0.5</span>,
                                                  min_det_gini_score <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>

<span class="co"># get top 2 genes per cluster and visualize with violinplot</span>
<span class="va">topgenes_gini</span> <span class="op">=</span> <span class="va">gini_markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span>
<span class="fu"><a href="../reference/violinPlot.html">violinPlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, genes <span class="op">=</span> <span class="va">topgenes_gini</span><span class="op">$</span><span class="va">feats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span>

<span class="co"># get top 6 genes per cluster and visualize with heatmap</span>
<span class="va">topgenes_gini2</span> <span class="op">=</span> <span class="va">gini_markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">6</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span>
<span class="fu"><a href="../reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, selected_genes <span class="op">=</span> <span class="va">topgenes_gini2</span><span class="op">$</span><span class="va">feats</span>,
                    metadata_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'leiden_clus'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cell-type-annotation">6. cell type annotation<a class="anchor" aria-label="anchor" href="#cell-type-annotation"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clusters_cell_types</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell A'</span>, <span class="st">'cell B'</span>, <span class="st">'cell C'</span>, <span class="st">'cell D'</span>,
                        <span class="st">'cell E'</span>, <span class="st">'cell F'</span>, <span class="st">'cell G'</span>, <span class="st">'cell H'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusters_cell_types</span><span class="op">)</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>
<span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/annotateGiotto.html">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, 
                              annotation_vector <span class="op">=</span> <span class="va">clusters_cell_types</span>, 
                              cluster_column <span class="op">=</span> <span class="st">'leiden_clus'</span>, 
                              name <span class="op">=</span> <span class="st">'cell_types'</span><span class="op">)</span>
<span class="co"># check new cell metadata</span>
<span class="fu"><a href="../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span>

<span class="co"># visualize annotations</span>
<span class="fu"><a href="../reference/spatDimPlot.html">spatDimPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, 
            spat_point_size <span class="op">=</span> <span class="fl">3</span>, dim_point_size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>

<span class="co"># heatmap</span>
<span class="va">topgenes_heatmap</span> <span class="op">=</span> <span class="va">gini_markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">4</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span>
<span class="fu"><a href="../reference/plotHeatmap.html">plotHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
            genes <span class="op">=</span> <span class="va">topgenes_heatmap</span><span class="op">$</span><span class="va">feats</span>, 
            gene_order <span class="op">=</span> <span class="st">'custom'</span>,
            gene_custom_order <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">topgenes_heatmap</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span>,
            cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,
            legend_nrows <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-grid">7. spatial grid<a class="anchor" aria-label="anchor" href="#spatial-grid"></a>
</h2>
<p>Create a grid based on defined stepsizes in the x,y(,z) axes.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seqfish_mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSpatialGrid.html">createSpatialGrid</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                              sdimx_stepsize <span class="op">=</span> <span class="fl">300</span>,
                              sdimy_stepsize <span class="op">=</span> <span class="fl">300</span>,
                              minimum_padding <span class="op">=</span> <span class="fl">50</span><span class="op">)</span>
<span class="fu">showGrids</span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span>

<span class="co"># visualize grid</span>
<span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, show_grid <span class="op">=</span> <span class="cn">T</span>, point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-network">8. spatial network<a class="anchor" aria-label="anchor" href="#spatial-network"></a>
</h2>
<ul>
<li>visualize information about the default Delaunay network<br>
</li>
<li>create a spatial Delaunay network (default)<br>
</li>
<li>create a spatial kNN network</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotStatDelaunayNetwork.html">plotStatDelaunayNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, maximum_distance <span class="op">=</span> <span class="fl">400</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, minimum_k <span class="op">=</span> <span class="fl">2</span>, 
                                    maximum_distance_delaunay <span class="op">=</span> <span class="fl">400</span><span class="op">)</span>
<span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, minimum_k <span class="op">=</span> <span class="fl">2</span>, 
                                    method <span class="op">=</span> <span class="st">'kNN'</span>, k <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="fu">showNetworks</span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span>

<span class="co"># visualize the two different spatial networks  </span>
<span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, show_network <span class="op">=</span> <span class="cn">T</span>,
         network_color <span class="op">=</span> <span class="st">'blue'</span>, spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,
         point_size <span class="op">=</span> <span class="fl">2.5</span>, cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span>

<span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, show_network <span class="op">=</span> <span class="cn">T</span>,
         network_color <span class="op">=</span> <span class="st">'blue'</span>, spatial_network_name <span class="op">=</span> <span class="st">'kNN_network'</span>,
         point_size <span class="op">=</span> <span class="fl">2.5</span>, cell_color <span class="op">=</span> <span class="st">'leiden_clus'</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-genes">9. spatial genes<a class="anchor" aria-label="anchor" href="#spatial-genes"></a>
</h2>
<p>Identify spatial genes with 3 different methods:<br>
- binSpect with kmeans binarization (default)<br>
- binSpect with rank binarization<br>
- silhouetteRank</p>
<p>Visualize top 4 genes per method.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">km_spatialgenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span>
<span class="fu">spatGenePlot</span><span class="op">(</span><span class="va">seqfish_mini</span>, expression_values <span class="op">=</span> <span class="st">'scaled'</span>, 
             genes <span class="op">=</span> <span class="va">km_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,
             point_shape <span class="op">=</span> <span class="st">'border'</span>, point_border_stroke <span class="op">=</span> <span class="fl">0.1</span>,
             show_network <span class="op">=</span> <span class="cn">F</span>, network_color <span class="op">=</span> <span class="st">'lightgrey'</span>, point_size <span class="op">=</span> <span class="fl">2.5</span>,
             cow_n_col <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="va">rank_spatialgenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/binSpect.html">binSpect</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, bin_method <span class="op">=</span> <span class="st">'rank'</span><span class="op">)</span>
<span class="fu">spatGenePlot</span><span class="op">(</span><span class="va">seqfish_mini</span>, expression_values <span class="op">=</span> <span class="st">'scaled'</span>, 
             genes <span class="op">=</span> <span class="va">rank_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>,
             point_shape <span class="op">=</span> <span class="st">'border'</span>, point_border_stroke <span class="op">=</span> <span class="fl">0.1</span>,
             show_network <span class="op">=</span> <span class="cn">F</span>, network_color <span class="op">=</span> <span class="st">'lightgrey'</span>, point_size <span class="op">=</span> <span class="fl">2.5</span>,
             cow_n_col <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="va">silh_spatialgenes</span> <span class="op">=</span> <span class="fu"><a href="../reference/silhouetteRank.html">silhouetteRank</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span><span class="op">)</span> <span class="co"># TODO: suppress print output</span>
<span class="fu">spatGenePlot</span><span class="op">(</span><span class="va">seqfish_mini</span>, expression_values <span class="op">=</span> <span class="st">'scaled'</span>, 
             genes <span class="op">=</span> <span class="va">silh_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">genes</span>,
             point_shape <span class="op">=</span> <span class="st">'border'</span>, point_border_stroke <span class="op">=</span> <span class="fl">0.1</span>,
             show_network <span class="op">=</span> <span class="cn">F</span>, network_color <span class="op">=</span> <span class="st">'lightgrey'</span>, point_size <span class="op">=</span> <span class="fl">2.5</span>,
             cow_n_col <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-co-expression-patterns">10. spatial co-expression patterns<a class="anchor" aria-label="anchor" href="#spatial-co-expression-patterns"></a>
</h2>
<p>Identify robust spatial co-expression patterns using the spatial network or grid and a subset of individual spatial genes.<br>
1. calculate spatial correlation scores<br>
2. cluster correlation scores</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># 1. calculate spatial correlation scores </span>
<span class="va">ext_spatial_genes</span> <span class="op">=</span> <span class="va">km_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span>
<span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu">detectSpatialCorGenes</span><span class="op">(</span><span class="va">seqfish_mini</span>,
                                         method <span class="op">=</span> <span class="st">'network'</span>, 
                                         spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,
                                         subset_genes <span class="op">=</span> <span class="va">ext_spatial_genes</span><span class="op">)</span>

<span class="co"># 2. cluster correlation scores</span>
<span class="va">spat_cor_netw_DT</span> <span class="op">=</span> <span class="fu">clusterSpatialCorGenes</span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, 
                                          name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>, k <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>
<span class="fu">heatmSpatialCorGenes</span><span class="op">(</span><span class="va">seqfish_mini</span>, spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>, 
                     use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span><span class="op">)</span>

<span class="va">netw_ranks</span> <span class="op">=</span> <span class="fu"><a href="../reference/rankSpatialCorGroups.html">rankSpatialCorGroups</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, 
                                  spatCorObject <span class="op">=</span> <span class="va">spat_cor_netw_DT</span>, 
                                  use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span><span class="op">)</span>
<span class="va">top_netw_spat_cluster</span> <span class="op">=</span> <span class="fu">showSpatialCorGenes</span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, 
                                            use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>,
                                            selected_clusters <span class="op">=</span> <span class="fl">6</span>, 
                                            show_top_genes <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>

<span class="va">cluster_genes_DT</span> <span class="op">=</span> <span class="fu">showSpatialCorGenes</span><span class="op">(</span><span class="va">spat_cor_netw_DT</span>, 
                                       use_clus_name <span class="op">=</span> <span class="st">'spat_netw_clus'</span>,
                                       show_top_genes <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">cluster_genes</span> <span class="op">=</span> <span class="va">cluster_genes_DT</span><span class="op">$</span><span class="va">clus</span>; <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cluster_genes</span><span class="op">)</span> <span class="op">=</span> <span class="va">cluster_genes_DT</span><span class="op">$</span><span class="va">feat_ID</span>


<span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu">createMetagenes</span><span class="op">(</span><span class="va">seqfish_mini</span>,
                               gene_clusters <span class="op">=</span> <span class="va">cluster_genes</span>,
                               name <span class="op">=</span> <span class="st">'cluster_metagene'</span><span class="op">)</span>
<span class="fu"><a href="../reference/spatCellPlot.html">spatCellPlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,
             spat_enr_names <span class="op">=</span> <span class="st">'cluster_metagene'</span>,
             cell_annotation_values <span class="op">=</span> <span class="va">netw_ranks</span><span class="op">$</span><span class="va">clusters</span>,
             point_size <span class="op">=</span> <span class="fl">1.5</span>, cow_n_col <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-hmrf-domains">11. spatial HMRF domains<a class="anchor" aria-label="anchor" href="#spatial-hmrf-domains"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hmrf_folder</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">temp_dir</span>,<span class="st">'/'</span>,<span class="st">'11_HMRF/'</span><span class="op">)</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">hmrf_folder</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">hmrf_folder</span>, recursive <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>

<span class="co"># perform hmrf</span>
<span class="va">my_spatial_genes</span> <span class="op">=</span> <span class="va">km_spatialgenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span><span class="op">$</span><span class="va">genes</span>
<span class="va">HMRF_spatial_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/doHMRF.html">doHMRF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                            expression_values <span class="op">=</span> <span class="st">'scaled'</span>,
                            spatial_genes <span class="op">=</span> <span class="va">my_spatial_genes</span>,
                            spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,
                            k <span class="op">=</span> <span class="fl">9</span>,
                            betas <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">28</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,
                            output_folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">hmrf_folder</span>, <span class="st">'/'</span>, <span class="st">'Spatial_genes/SG_top100_k9_scaled'</span><span class="op">)</span><span class="op">)</span>

<span class="co"># check and select hmrf</span>
<span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">28</span>, <span class="fl">30</span>, by <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/viewHMRFresults2D.html">viewHMRFresults2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                    HMRFoutput <span class="op">=</span> <span class="va">HMRF_spatial_genes</span>,
                    k <span class="op">=</span> <span class="fl">9</span>, betas_to_view <span class="op">=</span> <span class="va">i</span>,
                    point_size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/addHMRF.html">addHMRF</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                  HMRFoutput <span class="op">=</span> <span class="va">HMRF_spatial_genes</span>,
                  k <span class="op">=</span> <span class="fl">9</span>, betas_to_add <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">28</span><span class="op">)</span>,
                  hmrf_name <span class="op">=</span> <span class="st">'HMRF'</span><span class="op">)</span>

<span class="co"># visualize selected hmrf result</span>
<span class="va">giotto_colors</span> <span class="op">=</span> <span class="fu">Giotto</span><span class="fu">:::</span><span class="fu"><a href="../reference/getDistinctColors.html">getDistinctColors</a></span><span class="op">(</span><span class="fl">9</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">giotto_colors</span><span class="op">)</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">9</span>
<span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, cell_color <span class="op">=</span> <span class="st">'HMRF_k9_b.28'</span>,
         point_size <span class="op">=</span> <span class="fl">3</span>, coord_fix_ratio <span class="op">=</span> <span class="fl">1</span>, cell_color_code <span class="op">=</span> <span class="va">giotto_colors</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cell-neighborhood-cell-typecell-type-interactions">12. cell neighborhood: cell-type/cell-type interactions<a class="anchor" aria-label="anchor" href="#cell-neighborhood-cell-typecell-type-interactions"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">2841</span><span class="op">)</span>
<span class="va">cell_proximities</span> <span class="op">=</span> <span class="fu"><a href="../reference/cellProximityEnrichment.html">cellProximityEnrichment</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                                           cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,
                                           spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,
                                           adjust_method <span class="op">=</span> <span class="st">'fdr'</span>,
                                           number_of_simulations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>
<span class="co"># barplot</span>
<span class="fu"><a href="../reference/cellProximityBarplot.html">cellProximityBarplot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, 
                     CPscore <span class="op">=</span> <span class="va">cell_proximities</span>, 
                     min_orig_ints <span class="op">=</span> <span class="fl">5</span>, min_sim_ints <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">## heatmap</span>
<span class="fu"><a href="../reference/cellProximityHeatmap.html">cellProximityHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, CPscore <span class="op">=</span> <span class="va">cell_proximities</span>, 
                     order_cell_types <span class="op">=</span> <span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span>,
                     color_breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1.5</span>, <span class="fl">0</span>, <span class="fl">1.5</span><span class="op">)</span>, 
                     color_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'blue'</span>, <span class="st">'white'</span>, <span class="st">'red'</span><span class="op">)</span><span class="op">)</span>

<span class="co"># network</span>
<span class="fu"><a href="../reference/cellProximityNetwork.html">cellProximityNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, CPscore <span class="op">=</span> <span class="va">cell_proximities</span>, 
                     remove_self_edges <span class="op">=</span> <span class="cn">T</span>, only_show_enrichment_edges <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>

<span class="co"># network with self-edges</span>
<span class="fu"><a href="../reference/cellProximityNetwork.html">cellProximityNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>, CPscore <span class="op">=</span> <span class="va">cell_proximities</span>,
                     remove_self_edges <span class="op">=</span> <span class="cn">F</span>, self_loop_strength <span class="op">=</span> <span class="fl">0.3</span>,
                     only_show_enrichment_edges <span class="op">=</span> <span class="cn">F</span>,
                     rescale_edge_weights <span class="op">=</span> <span class="cn">T</span>,
                     node_size <span class="op">=</span> <span class="fl">8</span>,
                     edge_weight_range_depletion <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,
                     edge_weight_range_enrichment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="visualization-of-specific-cell-types">visualization of specific cell types<a class="anchor" aria-label="anchor" href="#visualization-of-specific-cell-types"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Option 1</span>
<span class="va">spec_interaction</span> <span class="op">=</span> <span class="st">"cell D--cell F"</span>
<span class="fu">cellProximitySpatPlot2D</span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                        interaction_name <span class="op">=</span> <span class="va">spec_interaction</span>,
                        show_network <span class="op">=</span> <span class="cn">T</span>,
                        cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,
                        cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,
                        cell_color_code <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell D'</span> <span class="op">=</span> <span class="st">'lightblue'</span>, <span class="st">'cell F'</span> <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span>,
                        point_size_select <span class="op">=</span> <span class="fl">4</span>, point_size_other <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="co"># Option 2: create additional metadata</span>
<span class="va">seqfish_mini</span> <span class="op">=</span> <span class="fu"><a href="../reference/addCellIntMetadata.html">addCellIntMetadata</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,
                             spatial_network <span class="op">=</span> <span class="st">'Delaunay_network'</span>,
                             cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,
                             cell_interaction <span class="op">=</span> <span class="va">spec_interaction</span>,
                             name <span class="op">=</span> <span class="st">'D_F_interactions'</span><span class="op">)</span>
<span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, cell_color <span class="op">=</span> <span class="st">'D_F_interactions'</span>, legend_symbol_size <span class="op">=</span> <span class="fl">3</span>,
         select_cell_groups <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'other_cell D'</span>, <span class="st">'other_cell F'</span>, <span class="st">'select_cell D'</span>, <span class="st">'select_cell F'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cell-neighborhood-interaction-changed-genes">13. cell neighborhood: interaction changed genes<a class="anchor" aria-label="anchor" href="#cell-neighborhood-interaction-changed-genes"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">## select top 25th highest expressing genes</span>
<span class="va">gene_metadata</span> <span class="op">=</span> <span class="fu"><a href="../reference/fDataDT.html">fDataDT</a></span><span class="op">(</span><span class="va">seqfish_mini</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gene_metadata</span><span class="op">$</span><span class="va">nr_cells</span>, <span class="va">gene_metadata</span><span class="op">$</span><span class="va">mean_expr</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gene_metadata</span><span class="op">$</span><span class="va">nr_cells</span>, <span class="va">gene_metadata</span><span class="op">$</span><span class="va">mean_expr_det</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">gene_metadata</span><span class="op">$</span><span class="va">mean_expr_det</span><span class="op">)</span>
<span class="va">high_expressed_genes</span> <span class="op">=</span> <span class="va">gene_metadata</span><span class="op">[</span><span class="va">mean_expr_det</span> <span class="op">&gt;</span> <span class="fl">4</span><span class="op">]</span><span class="op">$</span><span class="va">gene_ID</span>

<span class="co">## identify genes that are associated with proximity to other cell types</span>
<span class="va">CPGscoresHighGenes</span> <span class="op">=</span>  <span class="fu">findCPG</span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                              selected_genes <span class="op">=</span> <span class="va">high_expressed_genes</span>,
                              spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,
                              cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,
                              diff_test <span class="op">=</span> <span class="st">'permutation'</span>,
                              adjust_method <span class="op">=</span> <span class="st">'fdr'</span>,
                              nr_permutations <span class="op">=</span> <span class="fl">500</span>,
                              do_parallel <span class="op">=</span> <span class="cn">T</span>, cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="co">## visualize all genes</span>
<span class="fu"><a href="../reference/plotCellProximityGenes.html">plotCellProximityGenes</a></span><span class="op">(</span><span class="va">seqfish_mini</span>, cpgObject <span class="op">=</span> <span class="va">CPGscoresHighGenes</span>, method <span class="op">=</span> <span class="st">'dotplot'</span><span class="op">)</span>

<span class="co">## filter genes</span>
<span class="va">CPGscoresFilt</span> <span class="op">=</span> <span class="fu">filterCPG</span><span class="op">(</span><span class="va">CPGscoresHighGenes</span>, min_cells <span class="op">=</span> <span class="fl">2</span>, min_int_cells <span class="op">=</span> <span class="fl">2</span>, min_fdr <span class="op">=</span> <span class="fl">0.1</span>,
                          min_spat_diff <span class="op">=</span> <span class="fl">0.1</span>, min_log2_fc <span class="op">=</span> <span class="fl">0.1</span>, min_zscore <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>

<span class="co">## visualize subset of interaction changed genes (ICGs)</span>
<span class="va">ICG_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Cpne2'</span>, <span class="st">'Scg3'</span>, <span class="st">'Cmtm3'</span>, <span class="st">'Cplx1'</span>, <span class="st">'Lingo1'</span><span class="op">)</span>
<span class="va">ICG_genes_types</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell E'</span>, <span class="st">'cell D'</span>, <span class="st">'cell D'</span>, <span class="st">'cell G'</span>, <span class="st">'cell E'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ICG_genes</span><span class="op">)</span> <span class="op">=</span> <span class="va">ICG_genes_types</span>

<span class="fu"><a href="../reference/plotICG.html">plotICG</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
        cpgObject <span class="op">=</span> <span class="va">CPGscoresHighGenes</span>,
        source_type <span class="op">=</span> <span class="st">'cell A'</span>,
        source_markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Csf1r'</span>, <span class="st">'Laptm5'</span><span class="op">)</span>,
        ICG_genes <span class="op">=</span> <span class="va">ICG_genes</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="cell-neighborhood-ligand-receptor-cell-cell-communication">14. cell neighborhood: ligand-receptor cell-cell communication<a class="anchor" aria-label="anchor" href="#cell-neighborhood-ligand-receptor-cell-cell-communication"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">LR_data</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mouse_ligand_receptors.txt"</span>, package <span class="op">=</span> <span class="st">'Giotto'</span><span class="op">)</span><span class="op">)</span>

<span class="va">LR_data</span><span class="op">[</span>, <span class="va">ligand_det</span> <span class="op">:=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">mouseLigand</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">seqfish_mini</span><span class="op">@</span><span class="va">feat_ID</span><span class="op">[[</span><span class="st">'rna'</span><span class="op">]</span><span class="op">]</span>, <span class="cn">T</span>, <span class="cn">F</span><span class="op">)</span><span class="op">]</span>
<span class="va">LR_data</span><span class="op">[</span>, <span class="va">receptor_det</span> <span class="op">:=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">mouseReceptor</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">seqfish_mini</span><span class="op">@</span><span class="va">feat_ID</span><span class="op">[[</span><span class="st">'rna'</span><span class="op">]</span><span class="op">]</span>, <span class="cn">T</span>, <span class="cn">F</span><span class="op">)</span><span class="op">]</span>
<span class="va">LR_data_det</span> <span class="op">=</span> <span class="va">LR_data</span><span class="op">[</span><span class="va">ligand_det</span> <span class="op">==</span> <span class="cn">T</span> <span class="op">&amp;</span> <span class="va">receptor_det</span> <span class="op">==</span> <span class="cn">T</span><span class="op">]</span>
<span class="va">select_ligands</span> <span class="op">=</span> <span class="va">LR_data_det</span><span class="op">$</span><span class="va">mouseLigand</span>
<span class="va">select_receptors</span> <span class="op">=</span> <span class="va">LR_data_det</span><span class="op">$</span><span class="va">mouseReceptor</span>


<span class="co">## get statistical significance of gene pair expression changes based on expression ##</span>
<span class="va">expr_only_scores</span> <span class="op">=</span> <span class="fu"><a href="../reference/exprCellCellcom.html">exprCellCellcom</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                                   cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,
                                   random_iter <span class="op">=</span> <span class="fl">50</span>,
                                   gene_set_1 <span class="op">=</span> <span class="va">select_ligands</span>,
                                   gene_set_2 <span class="op">=</span> <span class="va">select_receptors</span><span class="op">)</span>

<span class="co">## get statistical significance of gene pair expression changes upon cell-cell interaction</span>
<span class="va">spatial_all_scores</span> <span class="op">=</span> <span class="fu"><a href="../reference/spatCellCellcom.html">spatCellCellcom</a></span><span class="op">(</span><span class="va">seqfish_mini</span>,
                                     spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,
                                     cluster_column <span class="op">=</span> <span class="st">'cell_types'</span>,
                                     random_iter <span class="op">=</span> <span class="fl">50</span>,
                                     gene_set_1 <span class="op">=</span> <span class="va">select_ligands</span>,
                                     gene_set_2 <span class="op">=</span> <span class="va">select_receptors</span>,
                                     adjust_method <span class="op">=</span> <span class="st">'fdr'</span>,
                                     do_parallel <span class="op">=</span> <span class="cn">T</span>,
                                     cores <span class="op">=</span> <span class="fl">4</span>,
                                     verbose <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span>


<span class="co">## * plot communication scores ####</span>

<span class="co">## select top LR ##</span>
<span class="va">selected_spat</span> <span class="op">=</span> <span class="va">spatial_all_scores</span><span class="op">[</span><span class="va">p.adj</span> <span class="op">&lt;=</span> <span class="fl">0.5</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">log2fc</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.1</span> <span class="op">&amp;</span> <span class="va">lig_nr</span> <span class="op">&gt;=</span> <span class="fl">2</span> <span class="op">&amp;</span> <span class="va">rec_nr</span> <span class="op">&gt;=</span> <span class="fl">2</span><span class="op">]</span>
<span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">setorder</a></span><span class="op">(</span><span class="va">selected_spat</span>, <span class="op">-</span><span class="va">PI</span><span class="op">)</span>

<span class="va">top_LR_ints</span> <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">selected_spat</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">PI</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">$</span><span class="va">LR_comb</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">33</span><span class="op">]</span>
<span class="va">top_LR_cell_ints</span> <span class="op">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">selected_spat</span><span class="op">[</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">PI</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">$</span><span class="va">LR_cell_comb</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">33</span><span class="op">]</span>

<span class="fu"><a href="../reference/plotCCcomHeatmap.html">plotCCcomHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                 comScores <span class="op">=</span> <span class="va">spatial_all_scores</span>,
                 selected_LR <span class="op">=</span> <span class="va">top_LR_ints</span>,
                 selected_cell_LR <span class="op">=</span> <span class="va">top_LR_cell_ints</span>,
                 show <span class="op">=</span> <span class="st">'LR_expr'</span><span class="op">)</span>


<span class="fu"><a href="../reference/plotCCcomDotplot.html">plotCCcomDotplot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                 comScores <span class="op">=</span> <span class="va">spatial_all_scores</span>,
                 selected_LR <span class="op">=</span> <span class="va">top_LR_ints</span>,
                 selected_cell_LR <span class="op">=</span> <span class="va">top_LR_cell_ints</span>,
                 cluster_on <span class="op">=</span> <span class="st">'PI'</span><span class="op">)</span>



<span class="co">## * spatial vs rank ####</span>
<span class="va">comb_comm</span> <span class="op">=</span> <span class="fu"><a href="../reference/combCCcom.html">combCCcom</a></span><span class="op">(</span>spatialCC <span class="op">=</span> <span class="va">spatial_all_scores</span>,
                      exprCC <span class="op">=</span> <span class="va">expr_only_scores</span><span class="op">)</span>

<span class="co"># top differential activity levels for ligand receptor pairs</span>
<span class="fu"><a href="../reference/plotRankSpatvsExpr.html">plotRankSpatvsExpr</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                   <span class="va">comb_comm</span>,
                   expr_rnk_column <span class="op">=</span> <span class="st">'exprPI_rnk'</span>,
                   spat_rnk_column <span class="op">=</span> <span class="st">'spatPI_rnk'</span>,
                   midpoint <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>

<span class="co">## * recovery ####</span>
<span class="co">## predict maximum differential activity</span>
<span class="fu"><a href="../reference/plotRecovery.html">plotRecovery</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
             <span class="va">comb_comm</span>,
             expr_rnk_column <span class="op">=</span> <span class="st">'exprPI_rnk'</span>,
             spat_rnk_column <span class="op">=</span> <span class="st">'spatPI_rnk'</span>,
             ground_truth <span class="op">=</span> <span class="st">'spatial'</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="export-giotto-analyzer-to-viewer">15. export Giotto Analyzer to Viewer<a class="anchor" aria-label="anchor" href="#export-giotto-analyzer-to-viewer"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">viewer_folder</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">temp_dir</span>, <span class="st">'/'</span>, <span class="st">'Mouse_cortex_viewer'</span><span class="op">)</span>

<span class="co"># select annotations, reductions and expression values to view in Giotto Viewer</span>
<span class="fu"><a href="../reference/exportGiottoViewer.html">exportGiottoViewer</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">seqfish_mini</span>,
                   output_directory <span class="op">=</span> <span class="va">viewer_folder</span>,
                   factor_annotations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell_types'</span>,
                                          <span class="st">'leiden_clus'</span>,
                                          <span class="st">'HMRF_k9_b.28'</span><span class="op">)</span>,
                   numeric_annotations <span class="op">=</span> <span class="st">'total_expr'</span>,
                   dim_reductions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'umap'</span><span class="op">)</span>,
                   dim_reduction_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'umap'</span><span class="op">)</span>,
                   expression_values <span class="op">=</span> <span class="st">'scaled'</span>,
                   expression_rounding <span class="op">=</span> <span class="fl">3</span>,
                   overwrite_dir <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com/" class="external-link">Ruben Dries</a>, Jiaji George Chen, Natalie Del Rossi, Emma Kelley, Junxiang Xu, Guo-Cheng Yuan.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
